src/qttools/datastructures/dsbcoo.py

```diff
@@ -425,6 +497,35 @@ def spy(self) -> tuple[ArrayLike, ArrayLike]:
         """
         return self.rows, self.cols
 
+    def reduce_to(
+        self, rows: ArrayLike, cols: ArrayLike, block_sizes: ArrayLike
+    ) -> "DSBCOO":
+        """Create a reduced matrix to the given rows and columns.
+
+        Parameters
+        ----------
+        rows : array_like
+            The rows to keep.
+        cols : array_like
+            The columns to keep.
+        block_sizes : array_like
+            The size of the blocks in the new matrix.
+
+        Returns
+        -------
+        DSBCOO
+            The reduced matrix.
+
+        """
+        mask = self.calc_reduce_to_mask(rows, cols)
+        return DSBCOO(
+            data=self.data[..., mask],
+            rows=self.rows[mask],
+            cols=self.cols[mask],
+            block_sizes=block_sizes,
+            global_stack_shape=self.global_stack_shape,
+        )
+
```

Comment:
Not a 100% sure why we need this functionality (have not yet looked at the PR over in quatrex). I guess this is somehow used to coerce data from one sparsity pattern into another, so something like
```python
a.data[:] = b.reduce_to(*a.spy(), a.block_sizes).data
```
In any case, i think this could overlap a lot with what the _get_items/_set_items methods are supposed to do. Can you check if you could not also do something like this instead?
```python
a.data[:] = b[*a.spy()]
```

Reply:
Ah yes, they give the same thing. It's a lot slower though. Should I remove the reduce_to method?

Reply:
Yeah, they are a lot slower and are bad memory-wise in the current implementation (will address this in #50 very soon), but they do for example not require that the sparsity patterns match which could be useful :man_shrugging:

Reply:
Okey, I removed them now.